lib/characteristic.dart

```diff
@@ -76,4 +77,11 @@ class CharacteristicWithValue extends Characteristic with WithValue {
   ) : super.fromJson(jsonObject, service, manager) {
     value = base64Decode(jsonObject[_CharacteristicMetadata.value]);
   }
+
+  CharacteristicWithValue setTransactionId(String transactionId) {
```

Comment:
What about passing transactionId to characteristicResponse from native side and only filtering it in CharacteristicMixin by obtaining the transactionId from raw json? I think that characteristic entity shouldn't be aware of our "pending transactions" mechanism.

Reply:
to be strict, characteristic entity is not aware of transaction id but there is a DTO that contains characteristic along with value and transactionID. I introduced additional mixin to separate transactionId from the previous model. If I would follow your suggestion I would have to introduced addition DTO anyway.